stadtreinigung_hamburg/StadtreinigungHamburg.py

```diff
@@ -6,53 +8,85 @@
+                        'Sec-Fetch-Site': 'same-origin'
+                    }
+                )
+            if(r.status_code!=200): raise Exception(F"Fetching step1 failed: Status code was {r.status_code}!")
+
+            streets = r.json()
+            logger.debug(F"found {len(streets)} street(s)")
+
+            if(len(streets) == 0): raise StreetNotFoundException(F"street not found with name {street}.")
+
+            asId = streets[0]["asId"]
+            logger.debug(F"asId is {asId}")
 
-        r = session.get('https://www.stadtreinigung.hamburg/privatkunden/abfuhrkalender/', params=params)
+            hnIds = streets[0]["hnIds"]
+            logger.debug(F"there are {len(hnIds)} numbers in this street.")
+            if(len(hnIds) == 0): raise StreetNumberNotFoundException(F"The street has no numbers?!")
+
+            hn = next((item for item in hnIds if item["name"] == number), None)
+            if(hn == None):
+                logger.info("retrying house number search with casefold!")
+                hn = next((item for item in hnIds if item["name"].casefold() == number.casefold()), None)
+                
+            if(hn == None):
+                raise StreetNumberNotFoundException(F"The street has {len(hnIds)} house numbers, but {number} is none of them!")
+
+            hnId = hn["hnId"]
+            logger.debug(F"hnId is {hnId}")
+        
+
+        # now fetch the actual garbage pickups
+        r = session.get(F"https://www.stadtreinigung.hamburg/abfuhrkalender/?tx_srh_pickups%5Bstreet%5D={asId}&tx_srh_pickups%5Bhousenumber%5D={hnId}")
+        if(r.status_code!=200): raise Exception(F"Fetching step2 failed: Status code was {r.status_code}!")
+        
         collections = []
 
-        # check for errors
-        possible_err = r.html.find('div.adresse')
-        if(len(possible_err) > 0):
-            print(possible_err[0].text)
-            if("Straße" in possible_err[0].text):
-                raise StreetNotFoundException("street not found.")
-            elif("Hausnummer" in possible_err[0].text):
-                hausnummer_sel = r.html.find('#hausnummer')
-                if(len(hausnummer_sel) == 1):
-                    hnids = [(option.text, option.attrs['value']) for option in r.html.find('#hausnummer option')[1:]]
-                    raise StreetNumberNotFoundException("street number not found.", hnids)
-                raise StreetNumberNotFoundException("street number not found and recommendations were not available.", [])
-
-        uuid = r.html.find("input[name=asId]")[0].attrs["value"]
-        uuid += "-" + r.html.find("input[name=hnId]")[0].attrs["value"]
-
-        for tr in r.html.find("#abfuhrkalender table tr"):
+        uuid = F"{asId}-{hnId}"
+        logger.debug(F"uuid is {uuid}")
+        
+        rows = r.html.find("tbody tr")
+        logger.debug(F"found {len(rows)} garbage collection data rows.")
+        actualLocale = locale.setlocale(locale.LC_TIME, "de_DE") 
```

Comment:
That is the critical part, but I have currently no idea to parse "Do 27. Januar 2022" without it. Help?!